src/esp32/single_encoder.rs

```diff
@@ -0,0 +1,341 @@
+use embedded_hal::digital::v2::InputPin;
+
+use super::encoder::PulseStorage;
+use super::pin::PinExt;
+use super::pulse_counter::{get_unit, isr_install, isr_installed, isr_uninstall};
+
+use crate::common::encoder::{
+    Direction, Encoder, EncoderPosition, EncoderPositionType, EncoderSupportedRepresentations,
+    SingleEncoder,
+};
+
+use core::ffi::{c_short, c_ulong};
+use esp_idf_sys as espsys;
+use espsys::pcnt_channel_edge_action_t_PCNT_CHANNEL_EDGE_ACTION_DECREASE as pcnt_count_dec;
+use espsys::pcnt_channel_edge_action_t_PCNT_CHANNEL_EDGE_ACTION_INCREASE as pcnt_count_inc;
+use espsys::pcnt_channel_level_action_t_PCNT_CHANNEL_LEVEL_ACTION_KEEP as pcnt_mode_keep;
+use espsys::pcnt_channel_t_PCNT_CHANNEL_0 as pcnt_channel_0;
+use espsys::pcnt_config_t;
+use espsys::pcnt_evt_type_t_PCNT_EVT_H_LIM as pcnt_evt_h_lim;
+use espsys::pcnt_evt_type_t_PCNT_EVT_L_LIM as pcnt_evt_l_lim;
+use espsys::{esp, EspError, ESP_OK};
+
+use std::collections::BTreeMap;
+use std::sync::atomic::{AtomicI32, Ordering};
+use std::sync::{Arc, Mutex};
+
+use crate::common::status::Status;
+
+// TODO: Make configurable?
+const MAX_GLITCH_MICROSEC: u16 = 1;
```

Comment:
why did you end up needing this?

Reply:
when I had two encoders wired at the same time, I found that powering one encoder's associated motor was creating enough noise on the other encoder's signal to increment the count for both units instead of just the one. The glitch filter solves this problem.